dotnet/src/Planners/Planners.OpenAI/Stepwise/InitialPlanPrompt.txt

```diff
@@ -0,0 +1,13 @@
+Create a step-by-step plan to satisfy the user's goal, using only the available functions or your own capabilities.
+
+[AVAILABLE FUNCTIONS]
+
+{{$available_functions}}
```

Comment:
So the functions are given both in the prompt and through the `Functions` parameter of the RequestSettings? How come?

Reply:
This planner works in two "phases": (1) ask the model for a set of steps to solve the goal; (2) repeatedly ask the model to solve the next step (using function calling) until a final answer is found.

This is the prompt for phase 1. This request does NOT leverage function calling, as we don't want a function response yet -- just the set of steps we will need to follow. But it is important that the AI know what functions/capabilities are available to use, so that it can return a set of steps that is doable. Hence, including the function manual here.

The second phase is where function calling (and the Functions parameter in request settings) comes in. We begin the chat history with `StepPrompt.txt` as the system message, then repeatedly ask the model to fulfill the next step. Each of these requests includes `Functions` in the request settings, but does not include the function manual in the chat history.

Does that make sense?

Reply:
Yeah this helps, thank you!

I'm wondering if you considered implementing `CreatePlan` in such a way that the initial set of steps that will be attempted could be visible as part of plan creation.